test/helpers/test/expectEvent.test.js

```diff
@@ -228,5 +288,89 @@ describe('expectEvent', function () {
         should.Throw(() => expectEvent.inLogs(this.logs, 'Boolean', { value: false }));
       });
     });
+
+    describe('with events emitted by an indirectly called contract', function () {
+      beforeEach(async function () {
+        this.secondEmitter = await IndirectEventEmitter.new();
+
+        this.value = 'OpenZeppelin';
+        ({ logs: this.logs } = await this.emitter.emitStringAndEmitIndirectly(this.value, this.secondEmitter.address));
+      });
+
+      it('accepts events emitted by the directly called contract', function () {
+        expectEvent.inLogs(this.logs, 'String', { value: this.value });
+      });
+
+      it('throws when passing events emitted by the indirectly called contract', function () {
+        should.Throw(() => expectEvent.inLogs(this.logs, 'IndirectString', { value: this.value }));
+      });
+    });
+  });
+
+  describe('inTransaction', function () {
+    describe('when emitting from called contract and indirect calls', function () {
+      context('string value', function () {
+        beforeEach(async function () {
+          this.secondEmitter = await IndirectEventEmitter.new();
+
+          this.value = 'OpenZeppelin';
+          const receipt = await this.emitter.emitStringAndEmitIndirectly(this.value, this.secondEmitter.address);
+          this.txHash = receipt.tx;
+        });
+
+        context('with directly called contract', function () {
+          it('accepts emitted events with correct string', async function () {
+            await expectEvent.inTransaction(this.txHash, EventEmitter, 'String', { value: this.value });
+          });
+
+          it('throws if an unemitted event is requested', async function () {
+            await expectEvent.inTransaction(this.txHash, EventEmitter, 'UnemittedEvent', { value: this.value })
+              .should.be.rejected;
+          });
+
+          it('throws if an incorrect string is passed', async function () {
+            await expectEvent.inTransaction(this.txHash, EventEmitter, 'String', { value: 'ClosedZeppelin' })
+              .should.be.rejected;
```

Comment:
I see, this is where you're using `chai-as-promised`. Aren't we using `shouldFail` for this?

Reply:
`shouldFail` only exports `reverting` and friends for now (i.e. they expect a specific error message). I guess we could add a simple one for non-Ethereum promises, where we only care about rejection and not the reason. I wouldn't want our users to use it though.